lib/membrane_ffmpeg_video_filter/text_overlay.ex

```diff
@@ -0,0 +1,157 @@
+              ],
+              fontsize: [
+                type: :int,
+                description:
+                  "Size of the displayed font. If not set, ffmpeg's default value (16) is applied",
+                default: nil
+              ],
+              fontcolor: [
+                type: :binary,
+                description:
+                  "Choose font color according to the ffmpeg color syntax (https://ffmpeg.org/ffmpeg-utils.html#color-syntax).
+                  If not set, ffmpeg's default value (black) is applied",
+                default: nil
+              ],
+              fontfile: [
+                type: :binary,
+                description:
+                  "Path to the file with the desired font. If not set, default font fallback from fontconfig is used",
+                default: nil
+              ],
+              box?: [
+                type: :boolean,
+                description: "Set to true if a box is to be displayed behind the text",
+                default: false
+              ],
+              boxcolor: [
+                type: :binary,
+                description: "If the box? is set to true, display a box in the given color",
+                default: nil
+              ],
+              border?: [
+                type: :boolean,
+                description: "Set to true to display a gray border around letters",
+                default: false
+              ],
+              x: [
+                type: :atom,
+                spec: :left | :right | :center,
+                description: "Horizontal position of the displayed text",
+                default: :left
+              ],
+              y: [
+                type: :atom,
+                spec: :top | :bottom | :center,
+                description: "Vertical position of the displayed text",
+                default: :bottom
+              ]
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: {Raw, aligned: true}
+
+  def_output_pad :output,
+    caps: {Raw, aligned: true}
+
+  @impl true
+  def handle_init(options) do
+    state =
+      options
+      |> Map.from_struct()
+      |> Map.put(:native_state, nil)
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, _size, :buffers, _ctx, %{native_state: nil} = state) do
+    # Wait until we have native state (after receiving caps)
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _ctx, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %Buffer{payload: payload} = buffer,
+        _ctx,
+        %{native_state: native_state} = state
+      ) do
+    case Native.filter(payload, native_state) do
+      {:ok, frame} ->
+        buffer = [buffer: {:output, %{buffer | payload: frame}}]
+        {{:ok, buffer}, state}
+
+      {:error, reason} ->
+        {{:error, reason}, state}
+    end
+  end
+
+  @impl true
+  def handle_caps(
```

Comment:
We expect caps to arrive before any buffer to process, so I'd move this function before `handle_process`